sql/rowexec: TestUncertaintyErrorIsReturned failed

[(sql/rowexec).TestUncertaintyErrorIsReturned failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2415052&tab=buildLog) on [master@e533f1939c55c34470e1c09958fe9f8205b63805](https://github.com/cockroachdb/cockroach/commits/e533f1939c55c34470e1c09958fe9f8205b63805):

Fatal error:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1b0f2bc]

```

Stack:
```
goroutine 16578 [running]:
github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror.CatchVectorizedRuntimeError.func1(0xc00522ee28)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror/error.go:60 +0x6c5
panic(0x3d24460, 0x78315a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/server/serverpb.(*LatencyGetter).GetLatency(0xc000c9b200, 0x53690e0, 0xc004a1b2c0, 0x200000001, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.go:98 +0x10c
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupInput(0xc00419da20, 0x53690e0, 0xc004a1b2c0, 0xc0033f1b00, 0x0, 0x0, 0x0, 0x0, 0xc0025637a0, 0x3, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:780 +0x1361
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupFlow.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:1025 +0x6a1
github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror.CatchVectorizedRuntimeError(0xc004cfee38, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror/error.go:93 +0x5f
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupFlow(0xc00419da20, 0x53690e0, 0xc004a1b2c0, 0xc0033f1b00, 0xc0044c8700, 0x3, 0x4, 0x0, 0xc000abd130, 0x2a47aafe4af2333d, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:979 +0xd3
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Setup(0xc004434280, 0x53691a0, 0xc004cbc5d0, 0xc00444bf10, 0xc0017d5f00, 0xc0024fd680, 0x2a47aafe4af2333d, 0xb60abb6d22e62fb2, 0xc0048f5d40)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:248 +0x68b
github.com/cockroachdb/cockroach/pkg/sql/distsql.(*ServerImpl).setupFlow(0xc0024fd680, 0x53691a0, 0xc0017d5fb0, 0x0, 0xc002207ea0, 0xc00444bf00, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/server.go:320 +0x57e
github.com/cockroachdb/cockroach/pkg/sql/distsql.(*ServerImpl).SetupFlow(0xc0024fd680, 0x53691a0, 0xc004cbc420, 0xc00444bf00, 0xc0024fd680, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/server.go:534 +0x1fd
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb._DistSQL_SetupFlow_Handler.func1(0x53691a0, 0xc004cbc420, 0x415ec80, 0xc00444bf00, 0x0, 0x0, 0x360, 0x3)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/api.pb.go:505 +0x86
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1(0x53691a0, 0xc004cbc420, 0x415ec80, 0xc00444bf00, 0xc004b3d1c0, 0xc004b3d1e0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:125 +0x69e
google.golang.org/grpc.getChainUnaryHandler.func1(0x53691a0, 0xc004cbc420, 0x415ec80, 0xc00444bf00, 0xc004cff948, 0xc136a8, 0x4044d60, 0xc004a1b280)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0xe7
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1(0x53691a0, 0xc004cbc420, 0x415ec80, 0xc00444bf00, 0xc004b3d1c0, 0xc004a1b280, 0xc004a1b280, 0x4, 0xc001ed0280, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:176 +0xa8
google.golang.org/grpc.getChainUnaryHandler.func1(0x53691a0, 0xc004cbc420, 0x415ec80, 0xc00444bf00, 0x0, 0x7ff26fdf4f80, 0x0, 0xc003e222a0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0xe7
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor(0x53691a0, 0xc004cbc420, 0x415ec80, 0xc00444bf00, 0xc004b3d1c0, 0xc004a1b240, 0xc004cffa80, 0xc136a8, 0x4044d60, 0xc004a1b240)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:60 +0x8a
google.golang.org/grpc.chainUnaryServerInterceptors.func1(0x53691a0, 0xc004cbc420, 0x415ec80, 0xc00444bf00, 0xc004b3d1c0, 0xc004b3d1e0, 0xc0022a3b58, 0x5247e8, 0x405ee40, 0xc004cbc420)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:907 +0xca
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb._DistSQL_SetupFlow_Handler(0x422b100, 0xc0024fd680, 0x53691a0, 0xc004cbc420, 0xc003e22240, 0xc0022af180, 0x53691a0, 0xc004cbc420, 0xc0033db000, 0x634)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/api.pb.go:507 +0x14b
google.golang.org/grpc.(*Server).processUnaryRPC(0xc002168b60, 0x53e35e0, 0xc00183c300, 0xc00444be00, 0xc000c9b440, 0x7833f40, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1082 +0x4fd
google.golang.org/grpc.(*Server).handleStream(0xc002168b60, 0x53e35e0, 0xc00183c300, 0xc00444be00, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1405 +0xd25
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc003f206e0, 0xc002168b60, 0x53e35e0, 0xc00183c300, 0xc00444be00)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:746 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:744 +0xa1

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_JOIN_t_AS_t2_ON_t1.x_=_t2.x

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestUncertaintyErrorIsReturned PKG=./pkg/sql/rowexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestUncertaintyErrorIsReturned.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/rowexec).TestUncertaintyErrorIsReturned failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2415258&tab=buildLog) on [master@e533f1939c55c34470e1c09958fe9f8205b63805](https://github.com/cockroachdb/cockroach/commits/e533f1939c55c34470e1c09958fe9f8205b63805):

Fatal error:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1b0f10c]

```

Stack:
```
goroutine 16757 [running]:
github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror.CatchVectorizedRuntimeError.func1(0xc003e60e28)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror/error.go:60 +0x6c5
panic(0x3d114e0, 0x781a580)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/server/serverpb.(*LatencyGetter).GetLatency(0xc0045e27b0, 0x5356160, 0xc0033dbac0, 0x200000001, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.go:98 +0x10c
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupInput(0xc00334c6e0, 0x5356160, 0xc0033dbac0, 0xc001581560, 0x0, 0x0, 0x0, 0x0, 0xc0014240c0, 0x3, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:780 +0x1361
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupFlow.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:1025 +0x6a1
github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror.CatchVectorizedRuntimeError(0xc000372e38, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror/error.go:93 +0x5f
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupFlow(0xc00334c6e0, 0x5356160, 0xc0033dbac0, 0xc001581560, 0xc003bb8000, 0x3, 0x4, 0x0, 0xc004247130, 0xb54146506b5d2d93, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:979 +0xd3
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Setup(0xc00327c640, 0x5356220, 0xc003277200, 0xc00250a910, 0xc003f55800, 0xc002a20480, 0xb54146506b5d2d93, 0xb35642b1b82ad5bf, 0xc00327a900)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:248 +0x68b
github.com/cockroachdb/cockroach/pkg/sql/distsql.(*ServerImpl).setupFlow(0xc002a20480, 0x5356220, 0xc003f55890, 0x0, 0xc001788780, 0xc00250a900, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/server.go:320 +0x57e
github.com/cockroachdb/cockroach/pkg/sql/distsql.(*ServerImpl).SetupFlow(0xc002a20480, 0x5356220, 0xc003277050, 0xc00250a900, 0xc002a20480, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/server.go:534 +0x1fd
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb._DistSQL_SetupFlow_Handler.func1(0x5356220, 0xc003277050, 0x414bd00, 0xc00250a900, 0x0, 0x0, 0x360, 0x3)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/api.pb.go:505 +0x86
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1(0x5356220, 0xc003277050, 0x414bd00, 0xc00250a900, 0xc003b66140, 0xc003b66160, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:125 +0x69e
google.golang.org/grpc.getChainUnaryHandler.func1(0x5356220, 0xc003277050, 0x414bd00, 0xc00250a900, 0xc000373948, 0xc134f8, 0x4031de0, 0xc0033dba80)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0xe7
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1(0x5356220, 0xc003277050, 0x414bd00, 0xc00250a900, 0xc003b66140, 0xc0033dba80, 0xc0033dba80, 0x4, 0xc0013aed20, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:176 +0xa8
google.golang.org/grpc.getChainUnaryHandler.func1(0x5356220, 0xc003277050, 0x414bd00, 0xc00250a900, 0x0, 0x7f0ddc81db28, 0x0, 0xc002b21620)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0xe7
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor(0x5356220, 0xc003277050, 0x414bd00, 0xc00250a900, 0xc003b66140, 0xc0033dba40, 0xc000373a80, 0xc134f8, 0x4031de0, 0xc0033dba40)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:60 +0x8a
google.golang.org/grpc.chainUnaryServerInterceptors.func1(0x5356220, 0xc003277050, 0x414bd00, 0xc00250a900, 0xc003b66140, 0xc003b66160, 0xc00352eb58, 0x524638, 0x404bec0, 0xc003277050)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:907 +0xca
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb._DistSQL_SetupFlow_Handler(0x4218180, 0xc002a20480, 0x5356220, 0xc003277050, 0xc002b215c0, 0xc0039201a0, 0x5356220, 0xc003277050, 0xc003b6a000, 0x634)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/api.pb.go:507 +0x14b
google.golang.org/grpc.(*Server).processUnaryRPC(0xc003b65ba0, 0x53d0660, 0xc000d0ef00, 0xc00250a800, 0xc0045e2900, 0x781cf20, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1082 +0x4fd
google.golang.org/grpc.(*Server).handleStream(0xc003b65ba0, 0x53d0660, 0xc000d0ef00, 0xc00250a800, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1405 +0xd25
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc002c859e0, 0xc003b65ba0, 0x53d0660, 0xc000d0ef00, 0xc00250a800)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:746 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:744 +0xa1

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_JOIN_t_AS_t2_ON_t1.x_=_t2.x

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestUncertaintyErrorIsReturned PKG=./pkg/sql/rowexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestUncertaintyErrorIsReturned.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/rowexec).TestUncertaintyErrorIsReturned failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2415591&tab=buildLog) on [master@2546fd425b86d8f36732ff69a3d3f14168107470](https://github.com/cockroachdb/cockroach/commits/2546fd425b86d8f36732ff69a3d3f14168107470):

Fatal error:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1b0fb5c]

```

Stack:
```
goroutine 16836 [running]:
github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror.CatchVectorizedRuntimeError.func1(0xc0068b0e28)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror/error.go:60 +0x6c5
panic(0x3d164e0, 0x7825590)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/server/serverpb.(*LatencyGetter).GetLatency(0xc0048c1200, 0x535adc0, 0xc0041e5dc0, 0x200000001, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.go:98 +0x10c
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupInput(0xc005d66dc0, 0x535adc0, 0xc0041e5dc0, 0xc0063fe120, 0x0, 0x0, 0x0, 0x0, 0xc003248960, 0x3, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:780 +0x1361
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupFlow.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:1025 +0x6a1
github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror.CatchVectorizedRuntimeError(0xc0034e8e38, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror/error.go:93 +0x5f
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupFlow(0xc005d66dc0, 0x535adc0, 0xc0041e5dc0, 0xc0063fe120, 0xc006836000, 0x3, 0x4, 0x0, 0xc000ac3b80, 0xef49712b8b413422, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:979 +0xd3
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Setup(0xc005fe94a0, 0x535ae80, 0xc005a5f560, 0xc0039e7710, 0xc003c49300, 0xc001da4d80, 0xef49712b8b413422, 0xfc63a8dc0a0a66b1, 0xc005d64240)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:248 +0x68b
github.com/cockroachdb/cockroach/pkg/sql/distsql.(*ServerImpl).setupFlow(0xc001da4d80, 0x535ae80, 0xc003c493b0, 0x0, 0xc004f201e0, 0xc0039e7700, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/server.go:320 +0x57e
github.com/cockroachdb/cockroach/pkg/sql/distsql.(*ServerImpl).SetupFlow(0xc001da4d80, 0x535ae80, 0xc005a5f3b0, 0xc0039e7700, 0xc001da4d80, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/server.go:534 +0x1fd
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb._DistSQL_SetupFlow_Handler.func1(0x535ae80, 0xc005a5f3b0, 0x4151820, 0xc0039e7700, 0x0, 0x0, 0x360, 0x3)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/api.pb.go:505 +0x86
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1(0x535ae80, 0xc005a5f3b0, 0x4151820, 0xc0039e7700, 0xc006364860, 0xc006364880, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:125 +0x69e
google.golang.org/grpc.getChainUnaryHandler.func1(0x535ae80, 0xc005a5f3b0, 0x4151820, 0xc0039e7700, 0xc0034e9948, 0xc134f8, 0x4037520, 0xc0041e5d80)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0xe7
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1(0x535ae80, 0xc005a5f3b0, 0x4151820, 0xc0039e7700, 0xc006364860, 0xc0041e5d80, 0xc0041e5d80, 0x4, 0xc00255d450, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:176 +0xa8
google.golang.org/grpc.getChainUnaryHandler.func1(0x535ae80, 0xc005a5f3b0, 0x4151820, 0xc0039e7700, 0x0, 0x7fc97cb501f0, 0x0, 0xc0058e0c00)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0xe7
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor(0x535ae80, 0xc005a5f3b0, 0x4151820, 0xc0039e7700, 0xc006364860, 0xc0041e5d40, 0xc0034e9a80, 0xc134f8, 0x4037520, 0xc0041e5d40)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:60 +0x8a
google.golang.org/grpc.chainUnaryServerInterceptors.func1(0x535ae80, 0xc005a5f3b0, 0x4151820, 0xc0039e7700, 0xc006364860, 0xc006364880, 0xc005bfab58, 0x524638, 0x40516e0, 0xc005a5f3b0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:907 +0xca
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb._DistSQL_SetupFlow_Handler(0x421db40, 0xc001da4d80, 0x535ae80, 0xc005a5f3b0, 0xc0058e0ba0, 0xc00226bca0, 0x535ae80, 0xc005a5f3b0, 0xc00681e000, 0x634)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/api.pb.go:507 +0x14b
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00311ab60, 0x53d5560, 0xc003ae8c00, 0xc0039e7600, 0xc0048c1350, 0x7827f30, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1082 +0x4fd
google.golang.org/grpc.(*Server).handleStream(0xc00311ab60, 0x53d5560, 0xc003ae8c00, 0xc0039e7600, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1405 +0xd25
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc002eee820, 0xc00311ab60, 0x53d5560, 0xc003ae8c00, 0xc0039e7600)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:746 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:744 +0xa1

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_JOIN_t_AS_t2_ON_t1.x_=_t2.x

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestUncertaintyErrorIsReturned PKG=./pkg/sql/rowexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestUncertaintyErrorIsReturned.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/rowexec).TestUncertaintyErrorIsReturned failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2415512&tab=buildLog) on [master@0653b921a3ea757000ec87a95a60522549ed89ea](https://github.com/cockroachdb/cockroach/commits/0653b921a3ea757000ec87a95a60522549ed89ea):

Fatal error:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x273a991]

```

Stack:
```
goroutine 17826 [running]:
github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror.CatchVectorizedRuntimeError.func1(0xc00d8de6b0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror/error.go:60 +0x8b0
panic(0x5b38fe0, 0xab3a6a0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/server/serverpb.(*LatencyGetter).GetLatency(0xc0087f9c20, 0x811e640, 0xc00c8c33c0, 0x300000001, 0x2)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/serverpb/status.go:98 +0x1b1
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupInput(0xc00cca6580, 0x811e640, 0xc00c8c33c0, 0xc004ffe240, 0x0, 0x0, 0x0, 0x0, 0xc00d2efc20, 0x3, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:780 +0x1c69
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupFlow.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:1025 +0xa69
github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror.CatchVectorizedRuntimeError(0xc00a7746f0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecbase/colexecerror/error.go:93 +0x71
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupFlow(0xc00cca6580, 0x811e640, 0xc00c8c33c0, 0xc004ffe240, 0xc00c816e00, 0x3, 0x4, 0x0, 0xc0095a17a0, 0x474d2323e90f196e, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:979 +0xfe
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Setup(0xc00c8132c0, 0x811e700, 0xc00cdb0c60, 0xc00b67fb10, 0xc0089bd700, 0xc00a39c6c0, 0x474d2323e90f196e, 0x13ff6bb04efec4a2, 0xc00d7b8000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:248 +0xd4e
github.com/cockroachdb/cockroach/pkg/sql/distsql.(*ServerImpl).setupFlow(0xc00a39c6c0, 0x811e700, 0xc0089bd710, 0x0, 0xc00413bae0, 0xc00b67fb00, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/server.go:320 +0x141a
github.com/cockroachdb/cockroach/pkg/sql/distsql.(*ServerImpl).SetupFlow(0xc00a39c6c0, 0x811e700, 0xc0089bd710, 0xc00b67fb00, 0xc00a39c6c0, 0x117e089, 0xc003f512a8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql/server.go:534 +0x2b0
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb._DistSQL_SetupFlow_Handler.func1(0x811e700, 0xc00cdb0ab0, 0x5f73dc0, 0xc00b67fb00, 0x0, 0x0, 0x0, 0xc00a775740)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/api.pb.go:505 +0xa2
github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1(0x811e700, 0xc00cdb0ab0, 0x5f73dc0, 0xc00b67fb00, 0xc009dae280, 0xc009dae2a0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:125 +0x78b
google.golang.org/grpc.getChainUnaryHandler.func1(0x811e700, 0xc00cdb0ab0, 0x5f73dc0, 0xc00b67fb00, 0x40, 0x5e59d40, 0x1, 0xc00c8c3380)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0x147
github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1(0x811e700, 0xc00cdb0ab0, 0x5f73dc0, 0xc00b67fb00, 0xc009dae280, 0xc00c8c3380, 0xc00c8c3380, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:176 +0x136
google.golang.org/grpc.getChainUnaryHandler.func1(0x811e700, 0xc00cdb0ab0, 0x5f73dc0, 0xc00b67fb00, 0x0, 0x38, 0x38, 0xc00a39c3d8)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921 +0x147
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor(0x811e700, 0xc00cdb0ab0, 0x5f73dc0, 0xc00b67fb00, 0xc009dae280, 0xc00c8c3340, 0xc00a775858, 0x101f159, 0xc00c8c3370, 0xc00c8c3340)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:60 +0xbf
google.golang.org/grpc.chainUnaryServerInterceptors.func1(0x811e700, 0xc00cdb0ab0, 0x5f73dc0, 0xc00b67fb00, 0xc009dae280, 0xc009dae2a0, 0xc000b28000, 0x319, 0xc00d273968, 0x438058)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:907 +0x111
github.com/cockroachdb/cockroach/pkg/sql/execinfrapb._DistSQL_SetupFlow_Handler(0x6040f20, 0xc00a39c6c0, 0x811e700, 0xc00cdb0ab0, 0xc00b55a060, 0xc002525ba0, 0x811e700, 0xc00cdb0ab0, 0xc00cbf5000, 0x632)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/execinfrapb/api.pb.go:507 +0x1d9
google.golang.org/grpc.(*Server).processUnaryRPC(0xc009c42680, 0x819dc20, 0xc003b4d380, 0xc0079d3d00, 0xc0062c4030, 0xab3d060, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1082 +0x955
google.golang.org/grpc.(*Server).handleStream(0xc009c42680, 0x819dc20, 0xc003b4d380, 0xc0079d3d00, 0x0)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1405 +0x12be
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0069e2550, 0xc009c42680, 0x819dc20, 0xc003b4d380, 0xc0079d3d00)
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:746 +0xc9
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:744 +0xb9

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestUncertaintyErrorIsReturned/vectorize=true/SELECT_*_FROM_t_AS_t1_JOIN_t_AS_t2_ON_t1.x_=_t2.x

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestUncertaintyErrorIsReturned PKG=./pkg/sql/rowexec TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestUncertaintyErrorIsReturned.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/server/serverpb/status.go
+++ b/pkg/server/serverpb/status.go
@@ -94,7 +94,11 @@ func (lg *LatencyGetter) GetLatency(
 	if lg.latencyMap == nil {
 		lg.latencyMap = make(map[roachpb.NodeID]map[roachpb.NodeID]int64)
 	}
-	response, _ := ss.Nodes(ctx, &NodesRequest{})
+	response, err := ss.Nodes(ctx, &NodesRequest{})
+	if err != nil {
+		// When latency is 0, it is not shown on EXPLAIN ANALYZE diagrams.
+		return 0
+	}
 	for _, sendingNode := range response.Nodes {
 		sendingNodeID := sendingNode.Desc.NodeID
 		if lg.latencyMap[sendingNodeID] == nil {

